### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by cwendling**
All this was mostly found by cppcheck while I tried to improve its output in #1415.  Not everything it reports is taken care of here, but a lot is.

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
No build issues, and I tested panel lockdown, disabling force-quit, changing panel backgrounds (both image and color), the panel run dialog, adding/removing clock/calendar locations, and tested wayland as well. The multimonitor behavior was same as before: panel on primary monitor in x11, and on wayland on whichever monitor a terminal it is called from is on or the leftmost monitor if called by the session.

This is but a quick test, but it looks good on my end and addresses lots of "papercut" bugs in the code itself. Lots to test, but would be more if the same work was split into a dozen PR's that have to be pulled separately.

File: mate-panel/panel-widget.c [hunk 1]
```diff
@@ -1693,7 +1690,7 @@ panel_widget_size_allocate(GtkWidget *widget, GtkAllocation *allocation)
 				prior_space = ad->constrained;
 				if (list->prev)
 				{
-					pad = list->prev->data;
+					AppletData *pad = list->prev->data;
```
Comment (raveit65) [inline mate-panel/panel-widget.c:1693]: This gives me a build warning in travis-ci (cppcheck) https://app.travis-ci.com/github/mate-desktop/mate-panel/jobs/613338178#L5099
```
mate-panel/panel-widget.c:1693:18: style: Variable 'pad' can be declared as pointer to const [constVariablePointer]

     AppletData *pad = list->prev->data;

                 ^
```

**Comment by raveit65** [comment]
Beside from the new warning it seems that the panel runs fine in quick test.

**Comment by lukefromdc** [comment]
What is the status of this? due to the number of file changes we may have to rewrite either this or other PR's to deal with merge conflicts, so I would prefer to have this finished and out of the way. The cleanups improve the quality of the code, and the cppcheck warning is but advice on how code could be further improved, not a build warning. Holding on merging this as dealing with that is an unanswered question at this time

File: mate-panel/panel-widget.c (hunk 1 above)
Comment (cwendling) [reply mate-panel/panel-widget.c]: Should be fixed (although it's kind of unrelated, it's good to have less mutables)

**Comment by cwendling** [comment]
@lukefromdc IMO it's ready.  I addressed @raveit65's comment although it's not really new from this PR, but my additional commit just marks a bunch of pointers const, so either the compiler will emit new warnings, or it's almost certainly free of behavioral changes.

**Comment by lukefromdc** [comment]
This works fine, save for not containing the pager hack which for obvious reasons in in a different PR

**Comment by lukefromdc** [comment]
Accidental closure on comment

**Pull request merged**